packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -146,6 +146,8 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
           ...globals.xcode.xcrunCommand(),
           'xcodebuild',
           '-exportArchive',
+          if (shouldCodesign) '-allowProvisioningDeviceRegistration',
+          if (shouldCodesign) '-allowProvisioningUpdates',
```

Comment:
Usually we combine into one if condition:
```suggestion
          if (shouldCodesign) ...<String>[
            '-allowProvisioningDeviceRegistration',
            '-allowProvisioningUpdates',
          ]
```